src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -1,27 +1,58 @@
 package mate.academy.spring.config;
 
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.service.impl.CustomUserDetailsService;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
-    @Autowired
-    public void configure(AuthenticationManagerBuilder auth) throws Exception {
-        auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("1@123").password(getEncoder().encode("123")).roles("USER");
+    private final CustomUserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+
+    public SecurityConfig(CustomUserDetailsService userDetailsService,
+                          PasswordEncoder passwordEncoder) {
+        this.userDetailsService = userDetailsService;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(userDetailsService).passwordEncoder(passwordEncoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .antMatchers(HttpMethod.POST,"/register").permitAll()
+                .antMatchers(HttpMethod.GET,"/cinema-halls")
+                .hasAnyRole(Role.RoleName.ADMIN.name(),
+                        Role.RoleName.USER.name())
+                .antMatchers(HttpMethod.POST,"/cinema-halls")
+                .hasAnyRole(Role.RoleName.ADMIN.name(),
+                        Role.RoleName.USER.name())
+                .antMatchers(HttpMethod.GET,"/movies").hasAnyRole(Role.RoleName.ADMIN.name(),
```

Comment:
For further please note that we can make static import on enum values and use shorter form in code:
e.g.: 

import static mate.academy.spring.model.Role.RoleName.ADMIN;
import static mate.academy.spring.model.Role.RoleName.USER;
....
.antMatchers(HttpMethod.GET,"/movies").hasAnyRole(ADMIN.name(), USER.name())